src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -1258,12 +1235,12 @@ final boolean tryLockPhase() {    // seqlock acquire
          */
         WorkQueue(ForkJoinWorkerThread owner, int id, int cfg,
                   boolean clearThreadLocals) {
-            if (clearThreadLocals)
-                cfg |= CLEAR_TLS;
-            this.config = cfg;
-            top = base = 1;
-            this.phase = id;
+            array = new ForkJoinTask<?>[owner == null ?
+                                        INITIAL_EXTERNAL_QUEUE_CAPACITY :
+                                        INITIAL_QUEUE_CAPACITY];
             this.owner = owner;
+            //            top = base = 1;
```

Comment:
Leftover?

Reply:
Yes, sorry for mis-committing incomplete-undo that needs another compensating change to avoid small performance glitch. Will fix.